graf2d/gpadv7/src/RPadBase.cxx

```diff
@@ -20,22 +20,37 @@
 
 using namespace std::string_literals;
 
-ROOT::Experimental::RPadBase::~RPadBase() = default;
+using namespace ROOT::Experimental;
+
+RPadBase::~RPadBase() = default;
 
 ///////////////////////////////////////////////////////////////////////////
 /// Use provided style for pad and all primitives inside
 
-void ROOT::Experimental::RPadBase::UseStyle(const std::shared_ptr<RStyle> &style)
+void RPadBase::UseStyle(const std::shared_ptr<RStyle> &style)
 {
    RDrawable::UseStyle(style);
    for (auto &drawable : fPrimitives)
       drawable->UseStyle(style);
 }
 
+///////////////////////////////////////////////////////////////////////////
+/// Add primitive
+
+void RPadBase::AddPrimitive(std::shared_ptr<RDrawable> drawable)
+{
+   if (drawable->GetCssType() == "pad") {
+      auto pad = dynamic_cast<RPad *>(drawable.get());
+      if (pad) pad->SetParent(this);
+   }
```

Comment:
Why not just
```suggestion
   if (auto pad = dynamic_cast<RPad *>(drawable.get())
      pad->SetParent(this);
   }
```
i.e. why do the extra CSS type string check?

Reply:
Yes, one can do dynamic cast directly.
Just if there are many sub-classes, such operation can be expensive compared to string compare.

Reply:
Let's not optimize this too much, we're anyway trying to avoid deep hierarchies. I vote for code simplicity! :-)